include/core/evaporation_model.h

```diff
@@ -0,0 +1,683 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 -  by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ */
+#ifndef lethe_evaporation_model_h
+#define lethe_evaporation_model_h
+
+#include <core/parameters.h>
+// only for the field definition
+#include <core/physical_property_model.h>
+
+using namespace dealii;
+
+/**
+ * @brief EvaporationModel. Abstract class that allows to calculate the
+ * evaporation mass, heat and momentum fluxes.
+ */
+class EvaporationModel
+{
+public:
+  EvaporationModel()
+  {
+    model_depends_on[temperature] = false;
+  }
+
+  /**
+   * @brief Instantiates and returns a pointer to a EvaporationModel
+   * object by casting it to the proper child class
+   *
+   * @param evaporation_parameters Evaporation model parameters
+   */
+  static std::shared_ptr<EvaporationModel>
+  model_cast(const Parameters::Evaporation &evaporation_parameters);
+
+
+  /**
+   * @brief Returns true if the EvaporationModel depends on a field, false if not.
+   */
+  inline bool
+  depends_on(const field &id)
+  {
+    return model_depends_on[id];
+  }
+
+  /**
+   * @brief mass_flux Calculates the value of the evaporation mass flux.
+   * @param fields_value Value of the various field on which the flux may depend.
+   * @return value of the flux calculated with the fields_value
+   */
+  virtual double
+  mass_flux(const std::map<field, double> &fields_value) = 0;
+
+  /**
+   * @brief vector_mass_flux Calculates the values of the evaporation mass flux
+   * for multiple points
+   * @param field_vectors Value of the various fields on which the flux may depend.
+   * @param mass_flux_vector Vectors of the mass flux values
+   */
+  virtual void
+  vector_mass_flux(const std::map<field, std::vector<double>> &field_vectors,
```

Comment:
I would consider overloading these functions and using the same name for the scalar and vector versions (but with different arguments). I think it improves readability.